lib/api/objectPutCopyPart.js

```diff
@@ -261,7 +262,12 @@ function objectPutCopyPart(authInfo, request, sourceBucket,
                     if (err && !err.NoSuchKey) {
                         log.debug('error getting current part (if any)',
                         { error: err });
-                        return next(err);
+                        // Make a best effort single attempt to cleanup the
+                        // orphaned data while returning the original error from
+                        // the metadata layer.
+                        return dataDelete(locations[0], log, () => {
```

Comment:
only one location to delete here?

Reply:
Yes, from what I saw with this particular operation. I was wondering though what the behavior is if the source data is originally from an MPU. i.e. whether the locations array will be more than one. In which case, the `dataDelete` function could be altered to accommodate that. I'll dig a bit deeper there.

Reply:
You can use the batch delete method instead if you have more than 1 location to delete.

Reply:
Found that indeed for copies where the source is an MPU object, there will be multiple locations to be cleaned up. I'm moving this over to use `batchDelete`.